### DISCUSSION TITLE
API call via Postman/SoapUI - Form data not valid

### DISCUSSION THREAD

**Discussion by sukapyx** [mosparo/mosparo#239] [Q&A] [answered]
Hello. 

I have Mosparo set up. When querying the API via SoapUI/Postman the request-submit-token request works fine:
![image](https://github.com/mosparo/mosparo/assets/143611950/d32e7bd1-e843-433d-b2eb-370b88a721b7)

When making the check-form-data request I get error for the form data not being valid. Constructed it as in documentation https://documentation.mosparo.io/docs/api/frontend#structure-of-formdata

Have also tried passing the array in different ways but no use (get 500). Cant figure out what is invalid in the data.
![image](https://github.com/mosparo/mosparo/assets/143611950/3719194f-dbda-4bb6-a6b6-ed26c4038b97)

[{"name": "name", "value": "Testing", "fieldPath": "input[text].name"},{"name": "emailAddress","value": "testing@testing.com","fieldPath": "input[email].emailAddress"},{"name": "message", "value": "testmessage", "fieldPath": "textarea.message"}]

BR

**Comment by zepich** [comment]
Hi @sukapyx 

Thank you very much for your question.

Can you enable the API debug mode in the project settings (https://documentation.mosparo.io/docs/api/api_debug_mode)? The API should then provide a more detailed explanation for the error in the response.

Please post the additional data so I can look into the problem.

Kind regards,

zepich

**Reply by sukapyx** [reply]
Hello!

Thank you for the reply!

API debug mode was enabled when doing this request. I unchecked, saved, rechecked and saved again, but still - no additional info.

![image](https://github.com/mosparo/mosparo/assets/143611950/fad11db7-f852-49e3-beeb-3bfcfd7df08c)

At first I was thinking maybe Im posting the array data somehow wrong, but all the other options eg. FormData[], FormData[].name etc result in server error. I have tried different ways, also using Postman but the result is the same.

BR

**Reply by zepich** [reply]
Hi @sukapyx 

I'm very sorry. I didn't check the error message well enough when I answered your question. We do not have additional debug information in that place (where the error happens).

Can you share this string as it is with me?

![image](https://github.com/mosparo/mosparo/assets/391914/a98f3cee-fd5a-457c-8ba5-12e9a73e7486)

I don't understand the issue right now. Your request looks correct, so there is no reason for this problem. When you send me the marked string above, I can test your string in my development environment and see what goes wrong.

Kind regards,

zepich

**Reply by sukapyx** [reply]
The string is: publicKey=GEmREErp0HTKPiwaGb4vDHvChRu6RcLT0fpvBRoWiOo&submitToken=pUaNK5wrwhP8hv5GFZF9lP-nrHyn2h6kpZOMaVgA0Rk&formData=%5B%7B%22name%22%3A%20%22name%22%2C%20%22value%22%3A%20%22Testing%22%2C%20%22fieldPath%22%3A%20%22input%5Btext%5D.name%22%7D%2C%7B%22name%22%3A%20%22emailAddress%22%2C%22value%22%3A%20%22testing%40testing.com%22%2C%22fieldPath%22%3A%20%22input%5Bemail%5D.emailAddress%22%7D%2C%7B%22name%22%3A%20%22message%22%2C%20%22value%22%3A%20%22testsonum%22%2C%20%22fieldPath%22%3A%20%22textarea.message%22%7D%5D

**Reply by zepich** [reply, marked as answer]
Hi @sukapyx 

Thank you very much!

I've tested your string with SoapUI and had the same issue. The problem is the structure of the `formData` parameter.

Generally, it is an object with the two properties `fields` and `ignoredFields`. These properties are an array with the form data objects in it.

Your string contains only the value of the `fields` property but not the `fields` and `ignoredFields` properties. So, the correct string looks like this:

```json
{"fields":[{"name": "name", "value": "Testing", "fieldPath": "input[text].name"},{"name": "emailAddress","value": "testing@testing.com","fieldPath": "input[email].emailAddress"},{"name": "message", "value": "testsonum", "fieldPath": "textarea.message"}],"ignoredFields":[]}
```

If you test this string, you should receive a successful response from the API.

Please let me know if that worked!

Kind regards,

zepich

**Reply by sukapyx** [reply]
Great, like that it works! Thank you.

![image](https://github.com/mosparo/mosparo/assets/143611950/464378b6-8536-4b4a-ad4d-8d246a5ae95d)

Sorry for the trouble, I based and constructed my test request purely on Mosparo documentation https://documentation.mosparo.io/docs/api/frontend#arguments-1 and lacked the knowledge, nor was able to find that information anywhere.

Best regards!

**Reply by zepich** [reply]
Hi @sukapyx 

Awesome! I'm happy that it's working now!

Absolutely no problem! Thank you for posting this question. You're right. It's missing in the documentation and I will update the documentation ASAP.

Thank you very much!

Kind regards,

zepich

**Reply by zepich** [reply]
Hi @sukapyx 

I've updated the documentation: https://documentation.mosparo.io/docs/api/frontend/#check-form-data

If you want, please have a look and tell me what you think about the adjusted documentation.

Kind regards,

zepich
